logstash-core/src/main/java/org/logstash/settings/StringSetting.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.logstash.settings;
+
+import java.util.Collections;
+import java.util.List;
+
+public class StringSetting extends BaseSetting<String> {
```

Comment:
The setting classes have migrated Boolean, Coercible and NullableString to java. Do you think we can keep this class as String instead of StringSetting?

Reply:
I tried to do that but then every instance of Java String needs to be referenced by its full classname.
In this case I think that having the suffix `Setting` make more explicit in the codebase to which class we are referring.
For Boolean was easier and less impacting. I think that for Logstash settings which names collides with Java standard type we could use the `Setting` postfix.
WDYT?

Reply:
oh... naming is hard. Could we use `SString` ? The first S stands for Setting. 
StringSetting sounds like referring a different thing comparing to Boolean. 
:face_exhaling: I can accept StringSetting if you think it is more comfortable

Reply:
Having a prefix for every class would be good, eg SettingBoolean, SettingString, SettingNullableString

Reply:
Ok I rename StringSetting to SettingString for this PR then I'll create a follow up PR for the others